aws-android-sdk-cognitoauth/src/main/java/com/amazonaws/mobileconnectors/cognitoauth/AuthClient.java

```diff
@@ -238,11 +254,16 @@ public void getTokens(final Uri uri) {
         if (uri == null) {
             return;
         }
-        // The flag
-        LocalDataManager.cacheHasReceivedRedirect(pool.awsKeyValueStore, context, pool.getAppId(), true);
         getTokens(uri, userHandler);
     }
 
+    /**
+     * Properly handles the event where the user cancels out of the custom tabs auth flow either by closing it
+     * or navigating back away from it.
+     */
+    public void handleCustomTabsCancelled() {
+        userHandler.onFailure(new AuthNavigationException("user cancelled"));
+    }
```

Comment:
why does this have to be exposed to the customers? Why do they need to use this method instead of implementing their own callback behavior?

Reply:
^ Same question for `handleFlowCancelled()`

Reply:
Question resolved in-person